### PR TITLE
fix documentation errors in build and add ECC Zoo links for product codes

### PR DESCRIPTION
**Pull request by Fe-r-oz**
I noticed there was some errors with build. For the latest documentation to be deployed, all these errors needed to be fixed. The above two errors were causes by the references. I have updated them.

I will also fix all the others errors so that the build passes. 

```julia
┌ Error: Entry tillich2009quantum is missing the year field(s).
└ @ BibInternal ~/.julia/packages/BibInternal/RsFF9/src/bibtex.jl:93
┌ Error: Entry kovalev1212quantum is missing the year field(s).
└ @ BibInternal ~/.julia/packages/BibInternal/RsFF9/src/bibtex.jl:93
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: skipped.
[ Info: ExpandTemplates: expanding markdown templates.
┌ Error: duplicate docs found for 'defining_set' in `@docs` block in src/Classical/cyclic_code.md:85-87
│ ```@docs
│ defining_set
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'set_distance_upper_bound!' in `@docs` block in src/Classical/linear_code.md:84-86
│ ```@docs
│ set_distance_upper_bound!
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: duplicate docs found for 'encode' in `@docs` block in src/Classical/linear_code.md:98-100
│ ```@docs
│ encode
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: unable to get the binding for 'CodingTheory.⊕' in `@docs` block in src/Classical/new_codes_from_old.md:25-27 from expression ':($(Expr(:incomplete, "incomplete: premature end of input")))' in module Main
│ ```@docs
│ CodingTheory.⊕
│ ```
│   exception = `binding` cannot understand expression `$(Expr(:incomplete, "incomplete: premature end of input"))`.
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: unable to get the binding for 'CodingTheory.×' in `@docs` block in src/Classical/new_codes_from_old.md:30-32 from expression ':($(Expr(:incomplete, "incomplete: premature end of input")))' in module Main
│ ```@docs
│ CodingTheory.×
│ ```
│   exception = `binding` cannot understand expression `$(Expr(:incomplete, "incomplete: premature end of input"))`.
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: unable to get the binding for 'CodingTheory.⊗' in `@docs` block in src/Classical/new_codes_from_old.md:35-37 from expression ':($(Expr(:incomplete, "incomplete: premature end of input")))' in module Main
│ ```@docs
│ CodingTheory.⊗
│ ```
│   exception = `binding` cannot understand expression `$(Expr(:incomplete, "incomplete: premature end of input"))`.
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: unable to get the binding for 'CodingTheory./' in `@docs` block in src/Classical/new_codes_from_old.md:44-46 from expression ':($(Expr(:incomplete, "incomplete: premature end of input")))' in module Main
│ ```@docs
│ CodingTheory./
│ ```
│   exception = `binding` cannot understand expression `$(Expr(:incomplete, "incomplete: premature end of input"))`.
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'plot_EXIT_chart' in `@docs` block in src/LDPC/analysis.md:30-32
│ ```@docs
│ plot_EXIT_chart
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'multiplicative_gap' in `@docs` block in src/LDPC/analysis.md:35-37
│ ```@docs
│ multiplicative_gap
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'irregular_LDPC_code' in `@docs` block in src/LDPC/codes.md:39-41
│ ```@docs
│ irregular_LDPC_code
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'shortest_cycles' in `@docs` block in src/LDPC/codes.md:119-121
│ ```@docs
│ shortest_cycles
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'ACE_spectrum' in `@docs` block in src/LDPC/codes.md:124-126
│ ```@docs
│ ACE_spectrum
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'shortest_cycle_ACE' in `@docs` block in src/LDPC/codes.md:128-130
│ ```@docs
│ shortest_cycle_ACE
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'ACE_distribution' in `@docs` block in src/LDPC/codes.md:132-134
│ ```@docs
│ ACE_distribution
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'average_ACE_distribution' in `@docs` block in src/LDPC/codes.md:136-138
│ ```@docs
│ average_ACE_distribution
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'median_ACE_distribution' in `@docs` block in src/LDPC/codes.md:140-142
│ ```@docs
│ median_ACE_distribution
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'mode_ACE_distribution' in `@docs` block in src/LDPC/codes.md:144-146
│ ```@docs
│ mode_ACE_distribution
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'find_MP_schedule' in `@docs` block in src/LDPC/decoders.md:24-26
│ ```@docs
│ find_MP_schedule
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
┌ Error: undefined binding 'decoder_simulation' in `@docs` block in src/LDPC/decoders.md:36-38
│ ```@docs
│ decoder_simulation
│ ```
└ @ Documenter ~/.julia/packages/Documenter/CJeWX/src/utilities/utilities.jl:44
[ Info: CollectCitations
┌ Error: The @cite link text "" does not match required regex
│   ast =
│    @ast MarkdownAST.Link("@cite", "")
│    
└ @ DocumenterCitations ~/.julia/packages/DocumenterCitations/OyiEp/src/citation_link.jl:151
┌ Error: Error collecting citations from "Tutorials/Message Passing.md"
...
```

### CODE REVIEW

**Comment by Fe-r-oz** [comment]
Please don't merge this yet. I am waiting the results from the build.

**Comment by Fe-r-oz** [comment]
I think this is ready for review. I had added the ECC Zoo links for product codes as well. For the false positives errors mentioned in https://github.com/esabo/CodingTheory/pull/40#issue-2534196646, I will look into it later. Thank you!

**Comment by esabo** [comment]
I'll just review it when the false positives are done because it won't compile before then anyway.

File: docs/src/Tutorials/Message Passing.md [hunk 1]
```diff
@@ -406,7 +406,7 @@ julia> Gallager_B(H, y)
 ```
 
 ## Decimation
-Decimation was previously used in message-passing applications outside of error correction and was applied to stabilizer codes in [](@cite). The idea is to freeze the value of a variable node. We can either do this from the start to obtain a so-called *genie-aided* decoder, or we can periodically pause message passing to fix a bit. In *guided decimation*, we pause every fixed number of rounds and freeze the value of the variable node with the highest log-likelihood ratio. In *automated decimation*, we pause after every iteration and fix any bit whose absolute value has passed a certain threshold.
```
Comment (esabo) [inline docs/src/Tutorials/Message Passing.md:409]: https://ieeexplore.ieee.org/abstract/document/10619083

**Review by esabo — CHANGES_REQUESTED** [review]

File: docs/src/Tutorials/Message Passing.md (hunk 1 above)
Comment (Fe-r-oz) [reply docs/src/Tutorials/Message Passing.md:409]: Thank you!

**Comment by Fe-r-oz** [comment]
Woohoo, documentation build passed. There is a way to suppress the errors using `checkdocs` in `makedocs`.  Using `@example` block can be helpful than importing .png to message passing.md. In the next PRs, I will do that as well. 

I think this is ready for review, Thank you!

**Comment by esabo** [comment]
There are a lot of deletions but I don't see things added back in?

**Comment by Fe-r-oz** [comment]
For some methods, docstrings are not defined, so called them in docs gives error. Namely: `decoder_simulation`, and `find_MP_schedule`. For some methods, docstring is defined, but calling it gives undefined binding. I will try to make sure to delete as less as possible. I can added the deleted methods using` [[` `]](@ref)` is possible.  The assets don't have the images that's why build was throwing error as well. I think `@example` block can be used in place of manually inputting images.

**Comment by Fe-r-oz** [comment]
Thanks for your comment! I hope this is okay. There are no png images in assets as that was causing the documentation error in `message passing.md` tutorial.  I have made sure to add the methods as `See: `or `See also:` so that user can refer to these methods. In the future, a lot more work on documentation can be done to revamp the `message passing.md `as well. 

This PR make sure that we can deploy the docs by suppressing all the errors for methods for which docstrings are not defined! I think it's ready for review, Thank you!

**Review by esabo — APPROVED** [review]

**Pull request merged**